nncf/torch/dynamic_graph/io_handling.py

```diff
@@ -74,44 +79,164 @@ def replicate_fn(tensor: torch.Tensor) -> torch.Tensor:
     return obj
 
 
-class InputInfoWrapManager:
-    INPUTS_MISMATCH_WARNING_TEXT = (
-        "Compression with regards to this input may occur incorrectly. Make sure "
-        "you call the compressed model with inputs that correspond to what NNCF was "
-        "configured to expect (either via NNCF config's input_infos, or custom"
-        "dummy_forward_fn/wrap_inputs_fn parameters), or that you know what you are "
-        "doing. This warning will not be shown again."
-    )
-    ARGS_INPUTS_MISMATCH_FORMAT_STRING = (
-        "Inputs mismatch - could not find arg with idx {} in NNCF-wrapped model "
-        "input args! " + INPUTS_MISMATCH_WARNING_TEXT
-    )
-    KWARGS_INPUTS_MISMATCH_FORMAT_STRING = (
-        "Inputs mismatch - could not find kwarg '{}' in NNCF-wrapped model input "
-        "kwargs! " + INPUTS_MISMATCH_WARNING_TEXT
-    )
+class ModelInputInfo(abc.ABC):
+    @abc.abstractmethod
+    def get_forward_inputs(self, device: str = None) -> Tuple[Tuple, Dict]:
+        pass
+
+
+class ModelInputElement(abc.ABC):
+    def __init__(self, keyword=None):
+        self.keyword = keyword
+
+    @abc.abstractmethod
+    def get_tensor_for_input(self) -> torch.Tensor:
+        pass
+
+    @abc.abstractmethod
+    def is_integer_input(self):
+        pass
+
+
+class FillerInputElement(ModelInputElement):
+    FILLER_TYPE_ONES = "ones"
+    FILLER_TYPE_ZEROS = "zeros"
+    FILLER_TYPE_RANDOM = "random"
+    FILLER_TYPES = [FILLER_TYPE_ONES, FILLER_TYPE_ZEROS, FILLER_TYPE_RANDOM]
+
+    def __init__(self, shape: List[int], type_str: str = "float", keyword=None, filler=None):
+        super().__init__(keyword=keyword)
+        self.shape = shape
+        self.type = self._string_to_torch_type(type_str)
+        if filler is None:
+            self.filler = self.FILLER_TYPE_ONES
+        else:
```

Comment:
Are other types of filler used anywhere?

Reply:
It's not us who decide this, but the model topology and its treatment of inputs. For example, you can't use zeros and random for LM inputs, but one can think of a model that would require non-uniform inputs (random will cover this) to trigger the main control flow path. I haven't encountered a model that would require zeros exactly, but it seems to be a natural complement to ones.